### PR TITLE
Add DBOS Cloud View

### PR DESCRIPTION
**Pull request by devhawk**
* add DBOS Cloud API invocation methods
* Add view for DBOS Cloud resources 
image

> Note, this PR adds new mechanism for accessing DBOS-Cloud resources (list of apps/dbs) *without* using `npx dbos-cloud`. Expectation is that later PRs will build on this new integration foundation to reduce/eliminate `npx dbos-cloud` dependency

### CODE REVIEW

File: resources/icon-stateful.svg
Comment (qianl15) [inline resources/icon-stateful.svg:1]: We don't need these icon files right?

Comment (devhawk) [reply resources/icon-stateful.svg:1]: we need dbos-logo.png but not the others. I tried attaching them to the view nodes and they didn't look good at this size. Already talking to AndyE about getting some in the future

File: src/dbosCloudApi.ts
Comment (qianl15) [inline src/dbosCloudApi.ts:1]: Do you have some ideas on how to automatically test these functions? Or do we have to test it manuall?

Comment (devhawk) [reply src/dbosCloudApi.ts:1]: I'll take a look at the [vscode testing extensions docs](https://code.visualstudio.com/api/working-with-extensions/testing-extension)

File: src/cloudCli.ts [hunk 1]
```diff
@@ -2,48 +2,53 @@ import * as vscode from 'vscode';
...
-    const json = JSON.parse(stdout) as DbosCloudApp;
-    logger.debug("dbos_cloud_app_status", { folder: folder.uri.fsPath, stdout: json });
-    return json;
+    const response = JSON.parse(stdout) as DbosCloudApp;
+    logger.debug("dbosCloudAppStatus", { folder: folder.uri.fsPath, response });
+    return response;
+}
+
+export async function dbosCloudAppList(folder: vscode.WorkspaceFolder) {
+    const { stdout } = await execFile("npx", ["dbos-cloud", "application", "list", "--json"], {
```
Comment (qianl15) [inline src/cloudCli.ts:19]: Do we still need to use `dbos-cloud` CLI, or can we directly use `listApps` function in `dbosCloudApi.ts`?

Comment (devhawk) [reply src/cloudCli.ts:19]: Eventually, this code that invokes `npx dbos-cloud` will go away, but that will be a future PR

File: src/dbosCloudApi.ts
Comment (devhawk) [reply src/dbosCloudApi.ts:1]: For future ref, vscode-test *NOT* supported on WSL2: https://github.com/microsoft/vscode-test/issues/245

**Review by qianl15 — APPROVED** [review]
The view is really nice!

File: src/dbosCloudApi.ts
Comment (devhawk) [reply src/dbosCloudApi.ts:1]: tracking adding tests with #23

File: src/cloudCli.ts (hunk 1 above)
Comment (devhawk) [reply src/cloudCli.ts:19]: tracking removing dbos-cloud cli integration with #24

**Pull request merged**